sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3468,7 +3469,8 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
         s"CREATE TEMPORARY TABLE ...$asSelect, use CREATE TEMPORARY VIEW instead", ctx)
     }
 
-    val partitioning = partitionExpressions(partTransforms, partCols, ctx)
+    val partitioning =
+      partitionExpressions(partTransforms, partCols, ctx) ++ bucketSpec.map(_.asTransform)
     val tableSpec = TableSpec(bucketSpec, properties, provider, options, location, comment,
```

Comment:
do we still need bucket spec in `TableSpec`?

Reply:
Today we restore partition col names from the v2 partitioning, I think we can do the same for bucket spec.

Reply:
I removed `bucketSpec` from `TableSpec` and restored  partition col names in `CatalogV2Implicits.asPartitionColumns`. Need to change a few tests.